src/helpers/utils.ts

```diff
@@ -273,17 +269,12 @@ export function download(file: Blob, fileType: string, fileName: string) {
 }
 
 export function handleStatusRedirect(
-  status: string,
+  status: ApplicationStatus,
+  applicationType: ApplicationType,
   history: RouteComponentProps['history']
 ) {
-  switch (status) {
-    case SUBMITTED:
-      return history.push('/registration-closed?param=validate')
-    case CONFIRMED:
-      return window.location.replace('/home')
-    case DECLINED:
-      return history.push('/registration-closed')
-    default:
-      return history.push('/landing')
-  }
+  if (([ApplicationStatus.SUBMITTED, ApplicationStatus.DECLINED, ApplicationStatus.APPROVED].indexOf(status) === -1) && Object.values(ApplicationStatus).includes(status))
+    (applicationType === ApplicationType.INTERNAL) ? history.push('/landing') : history.push('/?overlay=consent_osp')
+  else if ([ApplicationStatus.SUBMITTED, ApplicationStatus.APPROVED, ApplicationStatus.DECLINED].indexOf(status) !== -1)      window.location.replace('/home')
```

Comment:
Lines 276 and 278 contain a lot of duplication. Write
`const isADS = [...].includes(status)`
and then use that value true or false in those lines.

Reply:
@oyo I made some changes and make it less complicated. Please recheck once